core/src/main/java/org/elasticsearch/common/unit/Fuzziness.java

```diff
@@ -121,7 +130,7 @@ public int asDistance() {
     }
 
     public int asDistance(String text) {
-        if (this == AUTO) { //AUTO
+        if (fuzziness.equals("AUTO")) { //AUTO
```

Comment:
I meant to leverage our own equals method rather than a string comparison. AUTO.equals(this) ?